component-samples/demo/scripts/fdo_toolkit.sh

```diff
@@ -0,0 +1,282 @@
+echo "IP.2 = 127.0.0.1" >> web-server.conf
+echo "IP.3 = 172.17.0.1" >> web-server.conf
+
+}
+
+generate_keys() {
+cd $1
+bash demo_ca.sh
+bash web_csr_req.sh
+bash user_csr_req.sh
+bash keys_gen.sh
+chmod +r secrets/server-key.pem
+cp -r secrets service.env ../manufacturer/
+cp -r secrets service.env ../rv/
+cp -r secrets service.env ../owner/
+cp -r secrets service.env ../aio/
+cp -r secrets ../db/
+}
+
+start_db() {
+cd $1/db
+if [ ! "$( docker container inspect -f '{{.State.Status}}' $container_name )" == "running" ]; then
+  sed -i 's+innodb_change_buffer_max_size = 25+#innodb_change_buffer_max_size = 25+g' custom/config-file.cnf
+  docker-compose up --build -d
+fi
+}
+
+start_aio() {
+cd $1/aio
+sed -i 's/  #- org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/  - org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/g' service.yml
+sed -i 's/  - org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/  #- org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/g' service.yml
+docker-compose up --build -d
+docker logs -f pri-fdo-aio
+}
+
+start_mfg() {
+cd $1/manufacturer
+docker-compose up --build -d
+}
+
+start_rv() {
+cd $1/rv
+sed -i 's/  #- org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/  - org.fidoalliance.fdo.protocol.UntrustedRendezvousAcceptFunction/g' service.yml
+sed -i 's/  - org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/  #- org.fidoalliance.fdo.protocol.db.TrustedRendezvousAcceptFunction/g' service.yml
+docker-compose up --build -d
+}
+
+start_owner() {
+cd $1/owner
+docker-compose up --build -d
+}
+
+add_docker_internal_mappings() {
+
+ if ! grep -q "host.docker.internal" /etc/hosts; then
+   echo '127.0.0.1 host.docker.internal' | sudo tee -a /etc/hosts
+ fi
+
+ echo "Added host.docker.internal mappings"
+}
+
+aio_e2e_setup() {
+  add_docker_internal_mappings
+  install_dependencies
+  docker_cleaner
+  download_and_unpack_binaries
+  update_scripts ~/pri_code_base/pri-fidoiot-v$FDO_RELEASE/scripts
+  generate_keys ~/pri_code_base/pri-fidoiot-v$FDO_RELEASE/scripts
+  start_db ~/pri_code_base/pri-fidoiot-v$FDO_RELEASE
+  sleep 10
+  start_aio ~/pri_code_base/pri-fidoiot-v$FDO_RELEASE
+}
+
+#update_rvinfo_aio() {
+  #ipaddress=`ip addr | grep 'state UP' -A2 | tail -n1 | awk '{print $2}' | cut -f1  -d'/'`
+  #cert_path="~/pri_code_base/pri-fidoiot-v$FDO_RELEASE/aio"
+  #auth_arg="--cacert ${cert_path}/ca-cert.pem --cert ${cert_path}/api-user.pem"
+  #ip={$1:-${ipaddress}}
+  #get_cert=$(curl ${auth_arg} --silent -w "%{http_code}\n" --location --request POST "https://${ip}:8080/api/v1/rvinfo?ip=$ip&rvprot=http -H 'Content-Type: text/plain')
+  #get_cert_code=$(tail -n1 <<< "$get_cert")
+  #if [ "$get_cert_code" = "200" ]; then
+  #  echo "Successfully updated rvinfo"
+  #fi
+#}
+
+
+
+# Main Execution script starts from here
+
+if [[ -z "${ipaddress}" ]]; then
+  ipaddress=`ip addr | grep 'state UP' -A2 | tail -n1 | awk '{print $2}' | cut -f1  -d'/'`
```

Comment:
Based on interface being used, this not might always return correct IP address , better to use following
```suggestion
  ipaddress=`ip route get 8.8.8.8 | grep -oP 'src \K[^ ]+'`
```